<a href="https://colab.research.google.com/github/daraadeniyi3116-create/ACCESS-6.0-SAP-Data-Science-Track/blob/main/WEEK_3_SESSION_4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
import os

if not os.path.exists('/content/drive'):
    drive.mount('/content/drive')
    print("Mounted!")
else:
    print("Drive is already mounted.")

Drive is already mounted.


In [ ]:
import pandas as pd
import numpy as np

df = pd.read_csv('/content/Warehouse_and_Retail_Sales.csv')


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 307645 entries, 0 to 307644
Data columns (total 9 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   YEAR              307645 non-null  int64  
 1   MONTH             307645 non-null  int64  
 2   SUPPLIER          307478 non-null  object 
 3   ITEM CODE         307645 non-null  object 
 4   ITEM DESCRIPTION  307645 non-null  object 
 5   ITEM TYPE         307644 non-null  object 
 6   RETAIL SALES      307642 non-null  float64
 7   RETAIL TRANSFERS  307645 non-null  float64
 8   WAREHOUSE SALES   307645 non-null  float64
dtypes: float64(3), int64(2), object(4)
memory usage: 21.1+ MB


In [ ]:
df.isnull().sum()

YEAR                  0
MONTH                 0
SUPPLIER            167
ITEM CODE             0
ITEM DESCRIPTION      0
ITEM TYPE             1
RETAIL SALES          3
RETAIL TRANSFERS      0
WAREHOUSE SALES       0
dtype: int64

In [ ]:
f"Duplicate Rows: {df.duplicated().sum()}"

'Duplicate Rows: 0'

In [ ]:
df.describe()

YEAR          MONTH   RETAIL SALES  RETAIL TRANSFERS  \
count  307645.000000  307645.000000  307642.000000     307645.000000   
mean     2018.438525       6.423862       7.024071          6.936465   
std         1.083061       3.461812      30.986238         30.237195   
min      2017.000000       1.000000      -6.490000        -38.490000   
25%      2017.000000       3.000000       0.000000          0.000000   
50%      2019.000000       7.000000       0.320000          0.000000   
75%      2019.000000       9.000000       3.267500          3.000000   
max      2020.000000      12.000000    2739.000000       1990.830000   

       WAREHOUSE SALES  
count    307645.000000  
mean         25.294597  
std         249.916798  
min       -7800.000000  
25%           0.000000  
50%           1.000000  
75%           5.000000  
max       18317.000000

In [ ]:
df.head()

YEAR  MONTH                           SUPPLIER ITEM CODE  \
0  2020      1  REPUBLIC NATIONAL DISTRIBUTING CO    100009   
1  2020      1                          PWSWN INC    100024   
2  2020      1            RELIABLE CHURCHILL LLLP      1001   
3  2020      1          LANTERNA DISTRIBUTORS INC    100145   
4  2020      1               DIONYSOS IMPORTS INC    100293   

                      ITEM DESCRIPTION ITEM TYPE  RETAIL SALES  \
0                  BOOTLEG RED - 750ML      WINE          0.00   
1            MOMENT DE PLAISIR - 750ML      WINE          0.00   
2  S SMITH ORGANIC PEAR CIDER - 18.7OZ      BEER          0.00   
3        SCHLINK HAUS KABINETT - 750ML      WINE          0.00   
4       SANTORINI GAVALA WHITE - 750ML      WINE          0.82   

   RETAIL TRANSFERS  WAREHOUSE SALES  
0               0.0              2.0  
1               1.0              4.0  
2               0.0              1.0  
3               0.0              1.0  
4               0.0              0.0

## Handling Missing Values


# Missing Values
1. **SUPPLIER**: 167 missing.

SOLUTION; Fill with "Unknown", since 167 is less that 0.05% of the data.

2. **RETAIL SALES**: 3 missing.

SOLUTION; Fill with Zero (0), since its better than remove the whole entry.

3. **ITEM TYPE**: 1 missing.

SOLUTION; Remove the entire entry, it is irrelevant.

In [ ]:
df['SUPPLIER'] = df['SUPPLIER'].fillna('Unknown')

df = df.dropna(subset=['ITEM TYPE'])

df['RETAIL SALES'] = df['RETAIL SALES'].fillna(0.0)


In [ ]:
print("Remaining missing values:", df.isnull().sum().sum())

Remaining missing values: 0


There are no Duplicates


In [ ]:
df['TOTAL_SALES'] = df['RETAIL SALES'] + df['WAREHOUSE SALES']

df['RETAIL_SHARE'] = np.where(df['TOTAL_SALES'] == 0, 0, df['RETAIL SALES'] / df['TOTAL_SALES'])

In [ ]:
df['NON_REVENUE_MOVE'] = df['RETAIL TRANSFERS'].abs()

In [ ]:
df[['ITEM TYPE', 'TOTAL_SALES', 'RETAIL_SHARE']].head()

ITEM TYPE  TOTAL_SALES  RETAIL_SHARE
0      WINE         2.00           0.0
1      WINE         4.00           0.0
2      BEER         1.00           0.0
3      WINE         1.00           0.0
4      WINE         0.82           1.0

In [ ]:
df.to_csv('cleaned_sales_data.csv', index=False)

print("✅ File saved successfully as 'cleaned_sales_data.csv'")

✅ File saved successfully as 'cleaned_sales_data.csv'
